src/interfaces/continent.ts

```diff
@@ -1,84 +1,60 @@
-import { ApiGetter } from "./api";
-import { Outfit } from "./player";
-
-export interface Population {
-  readonly NC: number;
-  readonly TR: number;
-  readonly VS: number;
-}
-
 /**
- * An object that represents all information about a continent
+ * Static base information to load and cache.
  */
-export interface ContinentDetails {
+export interface BaseInfo {
+  readonly id: number;
+  readonly continent_id: number;
   readonly name: string;
-  readonly population: Population;
-  /**
-   * The faction that locked the continent.
-   */
-  readonly locked_by: string | null;
-  readonly base_states: () => Promise<Array<BaseState>>;
+  readonly map_pos: [number, number]; // Map coordinate of the base icon & name
+  readonly type_id: number;
+  readonly type_name: string;
+  readonly resource_amount: number;
+  readonly resource_id?: number;
+  readonly resource_name?: string;
 }
 
-export interface ContinentView {
-  view_url: string;
-  details: ContinentDetails;
+export interface BaseStatus {
+  readonly id: number;
+  readonly server_id: number;
+  readonly population: Population;
+  readonly owning_faction?: number;
+  readonly owning_outfit?: number;
+  readonly held_since: number; // UTC timestamp of the last cap or reset
 }
 
-export type ContinentViews = Array<ContinentView>;
-export type Continents = Array<ContinentDetails>;
-
 /**
- * Convert a list of `ContinentDetails` to `ContinentView`
- * @param conts The ContinentDetails to add view URLs to
- * @param view_urls The URLs to add to `conts`
- * @throws The number of URLs does not match the number of continents
+ * Static continent data, loaded once and cached.
  */
-export const to_continent_view = (
-  conts: Array<ContinentDetails>,
-  view_urls: Array<string>
-): ContinentViews => {
-  if (conts.length !== view_urls.length)
-    throw new Error(
-      "The number of URLs does not match the number of continents"
-    );
-  return conts.map((cont, i) => ({ details: cont, view_url: view_urls[i] }));
-};
-
-export type faction = "NC" | "TR" | "VS" | "NS";
+export interface ContinentInfo {
+  readonly id: number;
+  readonly name: string;
+  readonly description: string;
+  readonly bases: Array<BaseInfo>;
```

Comment:
Does the "bases" endpoint need to exist here? The `BaseInfo` payload can be queried by continent ID, so this field is redundant.

If there are use cases that require continent info without base states, it might be better to remove this field.

Reply:
Removed the aforementioned endpoint for now as the use case for querying base info for single bases is nonexistant.